File: C++/프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    int n = arr.size();
    
    if (n == 0) return answer; // 빈 배열 처리
    
    answer.push_back(arr[0]); // 첫 번째 원소는 무조건 추가

    for (int i = 1; i < n; ++i) {
        if (arr[i] != arr[i - 1]) {
            answer.push_back(arr[i]);
        }
    }

    return answer;
}

File: Python3/프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    if not arr:
        return []

    result = [arr[0]]  # 첫 번째 원소는 항상 포함

    for i in range(1, len(arr)):
        if arr[i] != arr[i - 1]:
            result.append(arr[i])

    return result
